casper/src/main/scala/coop/rchain/casper/util/DagOperations.scala

```diff
@@ -109,7 +110,12 @@ object DagOperations {
 
     if (b1 == b2) {
       b1.pure[F]
-    } else {
+    } else
+    // This constraints are a temp solution, we need to make LCA calculation fast. For now it is assumed
+    // that when one block is `staleLatestMessageThreshold` behind another - it is LCA for this pair.
+    if (b1.blockNum - b2.blockNum > Estimator.depth) b2.pure[F]
+    else if (b2.blockNum - b1.blockNum > Estimator.depth) b1.pure[F]
```

Comment:
So this is just an optimization. It will not change the result because this very old block will be anyway returned as LCA. Is that right?

Reply:
@tgrospic yes, but in general case you never know what should be `Estimator.depth`. If we have some slashing conditions like `you have to propose at least each N blocks`, this condition can be justified.

Reply:
But even then you cannot be sure how temporary forking influence this. Looks like you don't have much options except traversing the DAG till you find one LCA. Thats why I'm thinking - might be we can keep a records for joint LCAs for different blocks and somehow incrementally update. But thats just a raw guess.

Reply:
https://uhra.herts.ac.uk/bitstream/handle/2299/12152/906535.pdf
https://java2blog.com/lca-k-ary-tree-osqrtheight/